### ISSUE TITLE
Spider-man 2000 

### ISSUE THREAD

**Issue by andercard0** [stenzek/duckstation#1571]
**Game details**
Spider-Man (USA) SLUS-00875 

**Description of the issue / bug**
As reported on discord by a user: 
Spider Man 2000 is having a bug in the Catch Venom level. In the end of the building scene, where Venom shatters some windows, Spider Man clips on the wall instead of following him.

**Controller Troubleshoot Report**
Not Related

**Steps to reproduce**
Load the game play until second venom's chace part, then after it goes off the first building as soon as spider-man got on the wall he got stuck.

![image](https://user-images.githubusercontent.com/43047877/106760241-57001600-6612-11eb-89bd-0fc5af896632.png)

**Enhancements information**
No enhancements at all activated.

**Hardware/software**
Ryzen 5 3600 | 32Gb | Rx 580 8Gigs 

Tested on D3D11

**Emulator version**
Latest - 0.1-3042-g7d036d6f

**Additional context**
Attached MC for test exactly where the issue is happening

[Spider-Man (USA)_1.zip](https://github.com/stenzek/duckstation/files/5919064/Spider-Man.USA._1.zip)

**Comment by foxsevent** [comment]
Yeah that's not a Ducko issue, it's just a game bug, happens on every other emulator and even on OG hardware, it can happen it few other places as well, like when entering the bank.
https://youtu.be/H1DGmd3WHE8?t=2290

**Comment by andercard0** [comment]
> Yeah that's not a Ducko issue, it's just a game bug, happens on every other emulator and even on OG hardware, it can happen it few other places as well, like when entering the bank.
> https://youtu.be/H1DGmd3WHE8?t=2290

Hm.. well, i'm not convinced if it is really recorded from a real console to be honest.

**Comment by Luke3357** [comment]
Made a quick video of how it rolls on real hardware. The bug did not happen in my tries. 
https://youtu.be/Kq67gw4U0L4

**Comment by foxsevent** [comment]
> > Yeah that's not a Ducko issue, it's just a game bug, happens on every other emulator and even on OG hardware, it can happen it few other places as well, like when entering the bank.
> > https://youtu.be/H1DGmd3WHE8?t=2290
> 
> Hm.. well, i'm not convinced if it is really recorded from a real console to be honest. But good you quoted this video so Sten can see where the issue happens.

That sadly is the thing you never know if it's the real thing or not.


> Made a quick video of how it rolls on real hardware. The bug did not happen in my tries.
> https://youtu.be/Kq67gw4U0L4

Fair enough and thanks for confirming, I dunno what causes this bug I only remember that was always there.

**Comment by Luke3357** [comment]
> Small edit
> Here you have the same bug but on the PC version.
> https://youtu.be/UBTxVCU4G4o
> 35:20
I can confirm that the bug was very common on the PC version. Most noticeable if it was run on modern hardware.

**Comment by Luke3357** [comment]
>Small edit
>Here you have the same bug but on the PC version.
>https://youtu.be/UBTxVCU4G4o
>35:20

I can confirm, this bug was very common on the PC version. Most noticeable if the game was running on more modern hardware.

**Comment by stenzek** [comment]
Might be worth checking this one on the preview version with the latest timing changes, to see if it affects it at all.

**Comment by foxsevent** [comment]
> Might be worth checking this one on the preview version with the latest timing changes, to see if it affects it at all.

Tested and somewhat fixed.
Testing with all enhancrments disabled.
Interpreter Spidey starts swing in the wrong direction.
ICACHE and recompiler 9/10 times correct behaviour but occasionally Spidey after swinging will land lower on the wall then he should or he'll take one step forward after landing in the correct spot.

Small note.
Using any overclocking will revert any of these to the original bug.

**Comment by andercard0** [comment]
Quick update on this a friend of mine tested the Dreamcast version and it does not happen there, mentioned that both PC and Playstation Portable (PSP) has this issue as well. 

Thus, a game bug!

But. he did mention that: Dreamcast version caps the frame pace at 15fps at that specific part / cutscene; Which personally think petty strange. 
So, he suggested if maybe, and if feasible too, do the same in Duckstation. If you think it did worth the try, if not I'd not have any more inputs in this and would close it, anyway, you decide what to do @stenzek or whoever have another input in regards of this issue feel free to update it!

**Comment by Phantasm90** [comment]
May i suggest you put force recompiler icache on in per game settings by default in the future? I've seen that it helps with the USA and Japansese version, altough it's with the PAL versions (Europe, France, Germany, Italy and Spain) that the bug kinda persist (Spidey would swing to the right side of the building alot of times)

**Comment by Phantasm90** [comment]
I recorded these videos for the sake of comparison, this is the european version

Level: **Catch Venom**

Mednafen

https://user-images.githubusercontent.com/37343312/115977242-89078100-a54c-11eb-9452-423d45c65206.mp4

Duckstation

https://user-images.githubusercontent.com/37343312/115977258-a9cfd680-a54c-11eb-929e-909f2a1d0195.mp4

Level: **The Lizard Maze**

Mednafen

https://user-images.githubusercontent.com/37343312/115977288-f61b1680-a54c-11eb-8a58-4ee60f20b470.mp4

Duckstation

https://user-images.githubusercontent.com/37343312/115977308-1c40b680-a54d-11eb-8d88-13d5f5187e73.mp4

**Comment by MojoJojoDojo** [comment]
I think I remember i got that bug on the PC version but not sure...
Or at least something similar.

**Comment by Phantasm90** [comment]
> ICACHE and recompiler 9/10 times correct behaviour but occasionally Spidey after swinging will land lower on the wall then he should or he'll take one step forward after landing in the correct spot.

Yeah, i tested the USA version again and the bug is inconsistent, sometimes Spidey will swing right ahead and sometimes he will not, he even swing to the right like the PAL version a couple of times

> Made a quick video of how it rolls on real hardware. The bug did not happen in my tries.
> https://youtu.be/Kq67gw4U0L4

Notice that when you can see Spider-man and Venom through the windows, Spidey is way too behind Venom than he should be

![Real Hardware](https://user-images.githubusercontent.com/37343312/116021370-75c4e600-a61e-11eb-80f8-8c5ba6323eef.png)

![Duckstation](https://user-images.githubusercontent.com/37343312/116021468-a147d080-a61e-11eb-8356-546e1f7a24ce.png)

**Comment by andercard0** [comment]
> > ICACHE and recompiler 9/10 times correct behaviour but occasionally Spidey after swinging will land lower on the wall then he should or he'll take one step forward after landing in the correct spot.
> 
> Yeah, i tested the USA version again and the bug is inconsistent, sometimes Spidey will swing right ahead and sometimes he will not, he even swings to the right like the PAL version a couple of times

Indeed, inconsistent and there is more, you've tested with this white and blue skin, if you do with the normal skin  the bug does not occurs, a friend made 2 vids on both cases. 


https://user-images.githubusercontent.com/43047877/116076872-2fe23f00-a66b-11eb-89e5-e469040e5d6c.mp4


https://user-images.githubusercontent.com/43047877/116076878-31136c00-a66b-11eb-846a-e654d1b0396b.mp4

**Comment by Phantasm90** [comment]
Like in #1862, one line of dialogue won't play on  the more recent build, in this case is the "Thanks Lizard" from Spidey in "The Lizard Maze" level (the USA version used to play it with Icache, but i tested several times and it didn't play)

https://user-images.githubusercontent.com/37343312/122837204-17d51780-d2ca-11eb-85dc-562272f2c8da.mp4
